store.go

```diff
@@ -1020,7 +1043,11 @@ func (s *store) getLayerStoreLocked() (rwLayerStore, error) {
 	if err := os.MkdirAll(rlpath, 0700); err != nil {
 		return nil, err
 	}
-	glpath := filepath.Join(s.graphRoot, driverPrefix+"layers")
+	imgStoreRoot := s.imageStoreDir
+	if imgStoreRoot == "" {
+		imgStoreRoot = s.graphRoot
+	}
+	glpath := filepath.Join(imgStoreRoot, driverPrefix+"layers")
```

Comment:
I think it's good, but just in case.  Any chance a symlink could sneak into one of these join commands that might cause havoc?

Reply:
It is same as original graphroot logic, even graphroot can accept a symlink, since its not a workload provided input i think it should not cause any issues :)